### PR TITLE
tests: Bluetooth: Mesh: reduce blob and dfu models bsim tests execution time

### PR DESCRIPTION
**Pull request by alxelax**
Improvements those have been done so far:
    1. General transfer size and block sizes are smaller for blob.
    2. Waiting time is shorter for some blob bsim tests.
    3. The long dfu srv persistence test suite has been split into two.
    4. Retransmission for unicast messages has been reduced till zero.
    Table with measurements. 
Screenshot from 2024-04-19 10-40-14
Measurement without improvement has been done on main commit 5dd1cdda3b69f0970466cc4ab232b2cd99dfa247.

### CODE REVIEW

File: tests/bsim/bluetooth/mesh/src/dfu_blob_common.c
Comment (omkar3141) [inline tests/bsim/bluetooth/mesh/src/dfu_blob_common.c:55]: This will make unicast SAR pretty unstable for multi segment messages.

**Review by omkar3141 — APPROVED** [review]
Looks fine.

File: tests/bsim/bluetooth/mesh/src/dfu_blob_common.c
Comment (alxelax) [reply tests/bsim/bluetooth/mesh/src/dfu_blob_common.c:55]: In the real life - yes, but not in bsim environment.
bsim environment is deterministic and without looses. It means 100% sent frames are received (if frame collision didn't happen).
Even more, if many repeats are configured for SAR it will make the test scenario more difficult. In scenario where a couple of devices participate and send something at the same time they create a lot of collisions especially for transport acks. Transport acks collide with retransmission very often.

To solve these collisions, there is another SAR parameter `unicast_retrans_without_prog_count` that is responsible for retransmission if the segment wasn't acked. This parameter is set to 2 for DFU and BLOB tests. It means collisions will be managed to some extent on transport layer.

For bsim tests the test scenarios require to not be overloaded with extra traffic that might help in real life. This is kind of tradeoff.

File: subsys/bluetooth/mesh/dfu_srv.c
Comment (PavelVPV) [inline subsys/bluetooth/mesh/dfu_srv.c:81]: We should perhaps remove the state storing at line 412 as well otherwise if response transmission failed and device rebooted, it will restore `BT_MESH_DFU_PHASE_APPLYING` instead of `BT_MESH_DFU_PHASE_VERIFY_OK`:
https://github.com/zephyrproject-rtos/zephyr/blob/4c2d27bcbec761f4bb78da27299eea6d305c3992/subsys/bluetooth/mesh/dfu_srv.c#L422-L423

`store_state` is called here anyway at line 100.

**Review by HaavardRei — COMMENTED** [review]
LGTM, I'll approve once Pavel's concern has been addressed :)

**Review by PavelVPV — DISMISSED** [review]

**Review by HaavardRei — DISMISSED** [review]

File: subsys/bluetooth/mesh/dfu_srv.c
Comment (PavelVPV) [inline subsys/bluetooth/mesh/dfu_srv.c:82]: Sorry, I missed that our target sample reboots right in the apply callback, so we need to store the state after the `if` but before the callback is called.

File: subsys/bluetooth/mesh/dfu_srv.c
Comment (HaavardRei) [inline subsys/bluetooth/mesh/dfu_srv.c:102]: Perhaps use LOG_WRN or LOG_ERR here? 

Also, perhaps add the error code in the last log message? (I see that this isn't generally done in the rest of the file, but could be an improvement!)

Comment (alxelax) [reply subsys/bluetooth/mesh/dfu_srv.c:102]: I do not think this is critical for stack to generate warning or even error here. Application or test aka application can manage its own errors by self. As a debug information, this should be sufficient.

**Review by PavelVPV — DISMISSED** [review]

File: subsys/bluetooth/mesh/dfu_srv.c
Comment (HaavardRei) [reply subsys/bluetooth/mesh/dfu_srv.c:102]: Ok!

**Review by HaavardRei — DISMISSED** [review]

**Review by HaavardRei — APPROVED** [review]

**Review by omkar3141 — APPROVED** [review]

**Comment by alxelax** [comment]
@PavelVPV, I guess it got stuck in review because they wait for your approval.

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**
